gen/tests/test_adminrouter_tls_conf.py

```diff
@@ -104,17 +254,44 @@ def test_default(self, new_arguments):
         This test is parametrized to demonstrate that having no configuration
         produces the same results as setting the config variable to `'false'`.
         """
-        protocols = self.supported_ssl_protocols(
+        protocols = self.supported_tls_protocols_ar_master(
             new_config_arguments=new_arguments,
         )
         assert protocols == ['TLSv1.1', 'TLSv1.2']
 
-    def test_enable(self):
+    @pytest.mark.parametrize(
+        'enabled,expected_protocols', [
+            (('false', 'false', 'true'), ['TLSv1.2']),
+            (('false', 'true', 'true'), ['TLSv1.1', 'TLSv1.2']),
+            (('true', 'true', 'true'), ['TLSv1', 'TLSv1.1', 'TLSv1.2']),
+            (('true', 'false', 'true'), ['TLSv1', 'TLSv1.2']),
+            (('true', 'false', 'false'), ['TLSv1']),
+        ]
```

Comment:
Well, quite a few of these combinations are not technically sound. That must be kept in mind -- not all of these need to "work". That is, this test can be considered overly fine-grained and might get in the way of developers in the future. Fine for now, but important to keep in mind.